lib/vagrant-libvirt/config.rb

```diff
@@ -901,6 +906,7 @@ def finalize!
         @tpm_type = 'passthrough' if @tpm_type == UNSET_VALUE
         @tpm_path = nil if @tpm_path == UNSET_VALUE
         @tpm_version = nil if @tpm_version == UNSET_VALUE
+        @sysinfo = {} if @sysinfo == UNSET_VALUE
```

Comment:
Given the hash is being exposed directly it means that users have no concept of what can and cannot be set in the hash, adding bad values will just be ignored but there is nothing to give them any helpful suggestions or alert them when what has been inputted will be ignored.

This comes back to the implicit validation that is currently in the create domain. I think better to move that to this class and make one of two choices:
* provide a wrapper that performs validation on hash at input
* move the code confirming valid sections and entries to the validation function to run after the various Vagrantfile's are merged (I'm leaning towards this and happy to help move the code if this makes sense to you)

Probably will also want to add an entry to the merge function to ensure any sysinfo settings provided by the box Vagrantfile are deep merged with the project Vagrantfile sysinfo settings. I think it can be done via the following as we know that there are only two levels and they are hashes so there is no need to use a very complex deep merge:
```
result.sysinfo = sysinfo.merge(other.sysinfo) { |_k, x, y| x.merge(y) }
```

Though it's probably worth checking if that is desirable behaviour. I'm guessing it's likely better than requiring needing to provide a complete sysinfo hash to override a single entry?